xtask/src/tasks/verify_size.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::Xtask;
+use anyhow::Context;
+use object::read::Object;
+use object::read::ObjectSection;
+use object::read::ObjectSymbol;
+use object::read::ObjectSymbolTable;
+use std::collections::HashMap;
+
+struct ExpectedSymbol<'a> {
+    name: &'a str,
+    start_symbol: &'a str,
+    end_symbol: &'a str,
+    size: u64,
+}
+
+impl<'a> ExpectedSymbol<'a> {
+    const fn new(name: &'a str, start_symbol: &'a str, end_symbol: &'a str, size: u64) -> Self {
+        Self {
+            name,
+            start_symbol,
+            end_symbol,
+            size,
+        }
+    }
+}
+
+struct ExpectedSection<'a> {
+    name: &'a str,
+    size: u64,
+}
+
+impl<'a> ExpectedSection<'a> {
+    const fn new(name: &'a str, size: u64) -> Self {
+        Self { name, size }
+    }
+}
+
+// Known target sizes
+
+const HCL_KERNEL_SHIP_NAME: &str = "hcl-kernel-ship";
```

Comment:
minor: can we share this somewhere so there isn't a local copy of the string in the other file?

Same with below.